### PR TITLE
APR-2894 already an employer provider code and coverage complete

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs [hunk 1]
```diff
@@ -712,28 +727,14 @@ public async Task<IActionResult> ChangeProviderRoute(ChangeProviderRouteViewMode
...
-
-                var modelErrors = ModelState.Values.SelectMany(v => v.Errors);
-                foreach (var modelError in modelErrors)
-                {
-                    model.ErrorMessages.Add(new ValidationErrorDetail
-                    {
-                        Field = "ChangeApplicationRoute",
-                        ErrorMessage = modelError.ErrorMessage
-                    });
+                    TempData[_employerProviderValidationMessage] = modelErrors.ErrorMessage;
```
Comment (DevenKShah) [inline src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs:734]: We should be returning view here as the model state is invalid, isn't it?

File: src/SFA.DAS.ApplyService.Web/Views/Roatp/ProviderAlreadyOnRegister.cshtml [hunk 2]
```diff
@@ -21,7 +21,12 @@
                     </h2>
                     <div class="govuk-error-summary__body">
                         <ul class="govuk-list govuk-error-summary__list">
-                            <li><a sfa-validation-for="ChangeApplicationRoute"></a></li>
+                            @foreach (var error in Model.ErrorMessages)
```
Comment (DevenKShah) [inline src/SFA.DAS.ApplyService.Web/Views/Roatp/ProviderAlreadyOnRegister.cshtml:24]: What warranted this change?

File: src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs:734]: that's what causes the post to get put into the browser history, so we are avoiding that in all the PRG stories.

File: src/SFA.DAS.ApplyService.Web/Views/Roatp/ProviderAlreadyOnRegister.cshtml (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web/Views/Roatp/ProviderAlreadyOnRegister.cshtml:24]: Validation stopped working, in that the top message wasn't being written.  I tried various methods to get it working using the current setup, but none worked, so went with this pattern (which is also present in various preamble pages)

File: src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs:734]: It follows the pattern established in APR-2893 of writing the message to tempData and redirecting back to the GET

**Pull request closed without merging**